beta/nncf/tensorflow/quantization/quantizers.py

```diff
@@ -20,17 +21,34 @@
 from beta.nncf.tensorflow.layers.data_layout import get_channel_axis
 from beta.nncf.tensorflow.layers.data_layout import get_channel_size
 from beta.nncf.tensorflow.layers.operation import NNCFOperation
-from beta.nncf.tensorflow.quantization.config import QuantizationMode
-from beta.nncf.tensorflow.quantization.config import QuantizerConfig
 from beta.nncf.tensorflow.quantization.functions import asymmetric_quantize
 from beta.nncf.tensorflow.quantization.functions import symmetric_quantize
+from nncf.common.quantization.structs import QuantizationMode, QuantizerConfig, QuantizerSpec
+
+
+class TFQuantizerSpec(QuantizerSpec):
+    def __init__(self, num_bits: int,
+                 mode: QuantizationMode,
+                 signedness_to_force: Optional[bool],
+                 narrow_range: bool,
+                 per_channel: bool):
+        super().__init__(num_bits, mode, signedness_to_force, narrow_range)
+        self.per_channel = per_channel
+
+    @classmethod
+    def from_config(cls, qconfig: QuantizerConfig, narrow_range: bool):
+        return cls(qconfig.num_bits,
+                   qconfig.mode,
+                   qconfig.signedness_to_force,
+                   narrow_range,
+                   qconfig.per_channel)
```

Comment:
@alexsu52 I checked once more and it seems there is no way around using `TFQuantizerSpec` - it uses `.per_channel` for purposes of determining final scale shape, while `PTQuantizerSpec` uses explicit `scale_shape` initializer parameter.

Reply:
PyTorch does not use `per_channel` for quantizer instantiation. Therefore the base `QuantizerSpec` class cannot contain `per_channel`.